### PR TITLE
Setup App: Replace custom picker API with tauri dialog package

### PR DESCRIPTION
**Pull request by vovakulikov**
Based on https://github.com/sourcegraph/sourcegraph/pull/51666

## Background 
As you may know, prior to this PR, we used a custom GQL handler that uses Apple script to call the native file picker/dialog UI. In this PR, we use a native picker with Tauri dialog API. 

## Test plan
- Check new local repositories setup UI page
- Check setup wizard local repositories step
- Check both UI in Mac and Windows (if you do have Windows machine)

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.17 kb) | 0.07% (+10.20 kb) 🔺 | 0.08% (+10.03 kb) 🔺 | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/add-tauri-dialog-api/compare-report.html;tab=live_object) for a full comparison between the commits 975ad16 and e068185 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/setup-wizard/components/local-repositories-step/queries.ts [hunk 1]
```diff
@@ -1,13 +1,5 @@
 import { gql } from '@apollo/client'
 
-export const GET_LOCAL_DIRECTORY_PATH = gql`
-    query GetLocalDirectoryPath {
-        localDirectoriesPicker {
-            paths
-        }
-    }
```
Comment (limitedmage) [inline client/web/src/setup-wizard/components/local-repositories-step/queries.ts:8]: Can we remove the backend for this query?

**Review by limitedmage — APPROVED** [review]
Much simpler than expected!! Very nice!

File: client/web/src/setup-wizard/components/local-repositories-step/hooks.ts [hunk 2]
```diff
@@ -221,25 +221,26 @@ export function useLocalRepositories({ paths, skip }: LocalRepositoriesInput): L
...
+        if (selected !== null) {
+            return [selected]
+        }
 
-                return data?.localDirectoriesPicker?.paths ?? []
-            }),
-        [queryPath]
-    )
+        return null
+    }, [])
```
Comment (fkling) [inline client/web/src/setup-wizard/components/local-repositories-step/hooks.ts:243]: Looks like this whole function could be defined outside the component? No need for `useCallback`.

Comment (vovakulikov) [reply client/web/src/setup-wizard/components/local-repositories-step/hooks.ts:243]: Yes, spent too much time with hooks.

**Pull request merged**